File: festival-web/src/main/java/cn/bdqfork/web/route/response/ResponseHandlerFactory.java
package cn.bdqfork.web.route.response;

import cn.bdqfork.core.util.StringUtils;
import cn.bdqfork.web.constant.ContentType;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

/**
 * @author bdq
 * @since 2020/2/15
 */
public class ResponseHandlerFactory {
    private Map<String, ResponseHandleStrategy> responseHandlerMap = new ConcurrentHashMap<>(16);

    public ResponseHandlerFactory() {
        registerResponseHandler(ContentType.PLAIN, new TextPlainResponseHandler());
        registerResponseHandler(ContentType.JSON, new JsonResponseHandler());
        registerResponseHandler(ContentType.XML, new XmlResponseHandler());
        registerResponseHandler(ContentType.HTML, new HtmlResponseHandler());
    }

    public void registerResponseHandler(String contentType, ResponseHandleStrategy responseHandleStrategy) {
        responseHandlerMap.put(contentType, responseHandleStrategy);
    }

    public ResponseHandleStrategy getResponseHandler(String contentType) {
        if (StringUtils.isEmpty(contentType) || !responseHandlerMap.containsKey(contentType)) {
            contentType = ContentType.JSON;
        }
        return responseHandlerMap.get(contentType);
    }
}

File: festival-kotlin-web/src/main/kotlin/cn/bdqfork/kotlin/web/route/response/ResponseHandlerFactory.kt
package cn.bdqfork.web.route.response

import cn.bdqfork.core.util.StringUtils
import cn.bdqfork.web.constant.ContentType
import java.util.concurrent.ConcurrentHashMap

/**
 * @author bdq
 * @since 2020/2/15
 */
class ResponseHandlerFactory {
    private val responseHandlerMap: MutableMap<String, ResponseHandleStrategy?> = ConcurrentHashMap(16)
    private fun registerResponseHandler(contentType: String, responseHandleStrategy: ResponseHandleStrategy?) {
        responseHandlerMap[contentType] = responseHandleStrategy
    }

    fun getResponseHandler(contentType: String?): ResponseHandleStrategy {
        if (StringUtils.isEmpty(contentType) || !responseHandlerMap.containsKey(contentType)) {
            return responseHandlerMap[ContentType.JSON]!!
        }
        return responseHandlerMap[contentType]!!
    }

    init {
        registerResponseHandler(ContentType.PLAIN, TextPlainResponseHandler())
        registerResponseHandler(ContentType.JSON, JsonResponseHandler())
    }
}